logging/list_logs_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+import (
+	"bytes"
+	"context"
+	"log"
+	"strings"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/logging"
+	"cloud.google.com/go/logging/logadmin"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+const logID = "test-list-logs"
+
+var projectID string
+
+func TestMain(m *testing.M) {
+	ctx := context.Background()
+
+	tc, ok := testutil.ContextMain(m)
+	if !ok {
+		log.Fatal("test project not set up properly")
+		return
+	}
+	projectID = tc.ProjectID
+
+	client, err := logging.NewClient(ctx, projectID)
+	if err != nil {
+		log.Fatalf("logging.NewClient(%q) failed: %v", projectID, err)
+	}
+	defer client.Close()
+
+	adminClient, err := logadmin.NewClient(ctx, projectID)
+	if err != nil {
+		log.Fatalf("logadmin.NewClient(%q) failed: %v", projectID, err)
+	}
+	defer adminClient.Close()
+
+	// Create a log
+	logger := client.Logger(logID)
+	logger.Log(logging.Entry{Payload: "create a log"})
+	if err := logger.Flush(); err != nil {
+		log.Fatalf("logger.Flush() failed: %v", err)
+	}
+	// Delete the log
+	defer adminClient.DeleteLog(ctx, logID)
+
+	m.Run()
+}
+
+func TestListLogs(t *testing.T) {
+	testutil.Retry(t, 6, 10*time.Second, func(r *testutil.R) {
+		buf := &bytes.Buffer{}
+		if err := listLogs(buf, projectID); err != nil {
+			r.Errorf("listLogs(%q) failed: %v", projectID, err)
+			return
+		}
+		if got, want := buf.String(), logID; !strings.Contains(got, want) {
```

Comment:
This conditional is a little awkward to read.
Please either inline `buf.String()` and `logID` (the error message makes the order clear, so we dont need the specific variables `got` and `want`), or do the variable assignments before the conditional.

Reply:
This test pattern is suggested and widely used within Google codebase. go/go-test-examples#simple

Reply:
Done